modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -59,6 +66,570 @@ static bool s_first = true;
 static int s_shards = 0;
 static std::mutex s_lmdbStartupLock;
 
+std::pair<uint32_t, uint32_t> LMDBBackend::getSchemaVersionAndShards(std::string& filename)
+{
+  // cerr << "getting schema version for path " << filename << endl;
+
+  uint32_t schemaversion;
+
+  int rc;
+  MDB_env* env = nullptr;
+  MDB_dbi dbi;
+  MDB_val key, data;
+  MDB_txn* txn = nullptr;
+
+  if ((rc = mdb_env_create(&env)) != 0) {
+    throw std::runtime_error("mdb_env_create failed");
+  }
+
+  if ((rc = mdb_env_set_maxdbs(env, 20)) != 0) {
```

Comment:
This limit is a bit low when the LMDB can also contain other DBIs.

Reply:
does that matter if pdns is not opening them?

Reply:
I'm missing something - a shard migration needs 2 dbis, a main db migration needs 9, both numbers are way smaller than 20. Multiple migrations might not fit, depending on what future migrations look like.

Reply:
If I understand correctly, a migration from v4 to v5 opens 17 DBIs:

- pdns
- 8 unversioned v4 names
- 8 versioned v5 names

A future migration would likely also use 8 new names, so of you would need to migrate from v4 to v6 in two steps in the same process, you would likely need 25 DBIs, unless you are careful to close the old ones after each migration.

Reply:
it's 1+4+4, not 1+8+8. Future migrations will likely be entirely separate functions, especially now that this migration is done in terms of the C API, and thus easy to keep standalone. And if not, we can up the number then :)

Reply:
I think you are forgetting the `_0` DBIs here?

Reply:
you are right! 17 it is. I'll add a comment.